### PR TITLE
Add new Firebase project management service.

### PR DESCRIPTION
**Pull request by nbegley**
Add app-scoped actions for Firebase apps within Firebase projects. Uses
Firebase Management REST APIs
(https://firebase.google.com/docs/projects/api/reference/rest/).

### CODE REVIEW

File: test/unit/project-management/android-app.spec.ts [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+      packageName: 'test-package-name',
+    };
+
+    const VALID_ANDROID_APP_METADATA: AndroidAppMetadata = {
+      resourceName: 'test-resource-name',
+      appId: APP_ID,
+      displayName: 'test-display-name',
+      projectId: 'test-project-id',
+      packageName: 'test-package-name',
+    };
```
[the remark this answers is not in the document]
Comment (nbegley) [reply test/unit/project-management/android-app.spec.ts]: Done.

File: test/unit/project-management/android-app.spec.ts [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+          .returns(Promise.reject(EXPECTED_ERROR));
+      stubs.push(stub);
+      return androidApp.addShaCertificate(certificateToAdd)
+          .should.eventually.be.rejected.and.equal(EXPECTED_ERROR);
+    });
+
+    it('should resolve on success', () => {
+      const stub = sinon
+          .stub(ProjectManagementRequestHandler.prototype, 'addAndroidShaCertificate')
+          .returns(Promise.resolve(null));
```
[the remark this answers is not in the document]
Comment (nbegley) [reply test/unit/project-management/android-app.spec.ts]: Can you clarify what you mean by that?

File: test/unit/project-management/ios-app.spec.ts [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+    const VALID_IOS_APP_METADATA_API_RESPONSE = {
+      name: 'test-resource-name',
+      appId: APP_ID,
+      displayName: 'test-display-name',
+      projectId: 'test-project-id',
+      bundleId: 'test-bundle-id',
+    };
+
+    const VALID_IOS_APP_METADATA: IosAppMetadata = {
```
[the remark this answers is not in the document]
Comment (nbegley) [reply test/unit/project-management/ios-app.spec.ts]: Done.

File: test/unit/project-management/ios-app.spec.ts [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+          .returns(Promise.reject(EXPECTED_ERROR));
+      stubs.push(stub);
+      return iosApp.setDisplayName(newDisplayName)
+          .should.eventually.be.rejected.and.equal(EXPECTED_ERROR);
+    });
+
+    it('should resolve on success', () => {
+      const stub = sinon
+          .stub(ProjectManagementRequestHandler.prototype, 'setIosDisplayName')
+          .returns(Promise.resolve(null));
```
[the remark this answers is not in the document]
Comment (nbegley) [reply test/unit/project-management/ios-app.spec.ts]: Done. It looks like Promise.resolve() works by itself (and passes the linter).

File: test/unit/project-management/android-app.spec.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (hiranya911) [reply test/unit/project-management/android-app.spec.ts]: I think something like:

```
const stub = sinon
          .stub(ProjectManagementRequestHandler.prototype, 'addAndroidShaCertificate')
          .resolves(null);
```

[the remark this answers is not in the document]
Comment (schmidt-sebastian) [reply test/unit/project-management/android-app.spec.ts]: Hm, I honestly don't remember what this was referring to. Feel free to ignore.

**Pull request merged**